src/pydvl/utils/utility.py

```diff
@@ -259,3 +272,112 @@ def __call__(self, indices: Iterable[int]) -> float:
     def data(self) -> Dataset:
         """Returns the wrapped utility's :class:`~pydvl.utils.dataset.Dataset`."""
         return self.utility.data
+
+
+class MinerGameUtility(Utility):
+    r"""Toy game utility that is used for testing and demonstration purposes.
+
+    Consider a group of n miners, who have discovered large bars of gold.
+
+    If two miners can carry one piece of gold,
+    then the payoff of a coalition $S$ is:
+
+    $${
+    v(S) = \left\{\begin{array}{lll}
+    \mid S \mid / 2 & \text{, if} & \mid S \mid \text{ is even} \\
+    ( \mid S \mid - 1)/2 & \text{, if} & \mid S \mid \text{ is odd}
+    \end{array}\right.
+    }$$
+
+    If there are more than two miners and there is an even number of miners,
+    then the core consists of the single payoff where each miner gets 1/2.
+
+    If there is an odd number of miners, then the core is empty.
+
+    Taken from: https://en.wikipedia.org/wiki/Core_(game_theory)
+
+    :param n_miners: Number of miners that participate in the game.
+    """
+
+    def __init__(self, n_miners: int, **kwargs):
+        if n_miners <= 2:
+            raise ValueError(f"n_miners, {n_miners} should be > 2")
+        self.n_miners = n_miners
+
+        x = np.arange(n_miners)[..., np.newaxis]
+        # The y values don't matter here
+        y = np.zeros_like(x)
+
+        self.data = Dataset(x_train=x, y_train=y, x_test=x, y_test=y)
+
+    def __call__(self, indices: Iterable[int]) -> float:
+        n = len(tuple(indices))
+        if n % 2 == 0:
+            return n / 2
+        else:
+            return (n - 1) / 2
```

Comment:
```suggestion
        return len(tuple(indices)) // 2
```

Reply:
To be honest, I find the version with the `if else` more readable.